datafusion/sqllogictest/test_files/unnest.slt

```diff
@@ -136,8 +150,43 @@ query error DataFusion error: Error during planning: unnest\(\) can only be appl
 select unnest(column3) from unnest_table;
 
 ## Multiple unnest functions in selection
-query error DataFusion error: This feature is not implemented: Only support single unnest expression for now
-select unnest(column1), unnest(column2) from unnest_table;
+query ?I
+select unnest([]), unnest(NULL::int[]);
```

Comment:
Is `NULL::int[]` difference from `NULL`?

Reply:
I think we should support null as well 🤔 

```
query error DataFusion error: This feature is not implemented: unnest\(\) does not support null yet
select unnest([]), unnest(NULL);
```

Reply:
I'm not sure how to support it yet.  PostgreSQL, BigQuery, and Clickhouse don't support unnesting untyped NULLs.
```sh
postgres=# select unnest(NULL);
ERROR:  function unnest(unknown) is not unique
LINE 1: select unnest(NULL);
               ^
HINT:  Could not choose a best candidate function. You might need to add explicit type casts.
```
Although DuckDB supports it, its behavior seems a bit strange to me.
```sh
D select unnest(NULL), unnest([1,2]);
┌──────────────┬───────────────────────────────┐
│ unnest(NULL) │ unnest(main.list_value(1, 2)) │
│    int32     │             int32             │
├──────────────────────────────────────────────┤
│                    0 rows                    │
└──────────────────────────────────────────────┘
D select unnest(NULL::int[]), unnest([1,2]);
┌─────────────────────────────────┬───────────────────────────────┐
│ unnest(CAST(NULL AS INTEGER[])) │ unnest(main.list_value(1, 2)) │
│              int32              │             int32             │
├─────────────────────────────────┼───────────────────────────────┤
│                                 │                             1 │
│                                 │                             2 │
└─────────────────────────────────┴───────────────────────────────┘
```
Unnesting an untyped NULL  and a typed NULL produced different results.

I think there may be two appropriate methods:
1. Use type coercion to convert an untype NULL into a certain fixed List type like `cast(NULL, NULL::int[])`.
2. Explicitly reject unnesting untyped NULLs.

Reply:
I prefer 2, since PostgresSQL, and Clickhouse return errors in this case, we follow the majority.

Reply:
> I prefer 2, since PostgresSQL, and Clickhouse return errors in this case, we follow the majority.

Make sense to me!

Reply:
I double checked and this PR does error as expected. I'll add a small test for this case here and merge this PR.

```
❯ select unnest(null) from t;
This feature is not implemented: unnest() does not support null yet
```